model/task/test_result.go

```diff
@@ -345,3 +365,116 @@ func sortTestResults(results []testresult.TestResult, opts *FilterOptions, baseS
+	prestoBucket, err := o.GetPrestoBucket(ctx, credentials)
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := prestoBucket.Get(ctx, t.PrestoPartitionKey())
+	if err != nil {
+		return nil, errors.Wrap(err, "getting Parquet test results")
+	}
+	defer func() {
+		err = r.Close()
+		grip.WarningWhen(err != nil, message.WrapError(err, message.Fields{
+			"message": "closing Presto test results bucket reader",
+		}))
+	}()
+	data, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading Parquet test results")
+	}
+
+	fr, err := goparquet.NewFileReader(bytes.NewReader(data))
+	if err != nil {
+		return nil, errors.Wrap(err, "creating Parquet reader")
+	}
+	pr := floor.NewReader(fr)
+	defer func() {
+		err = pr.Close()
+		grip.WarningWhen(err != nil, message.WrapError(err, message.Fields{
+			"message": "closing Parquet test results reader",
+		}))
+	}()
+
+	var parquetResults []testresult.ParquetTestResults
+	for pr.Next() {
+		row := testresult.ParquetTestResults{}
+		if err := pr.Scan(&row); err != nil {
+			return nil, errors.Wrap(err, "reading Parquet test results row")
+		}
+
+		parquetResults = append(parquetResults, row)
+	}
+
+	if err := pr.Err(); err != nil {
+		return nil, errors.Wrap(err, "reading Parquet test results rows")
+	}
+
+	var results []testresult.TestResult
+	for _, result := range parquetResults {
+		results = append(results, result.ConvertToTestResultSlice()...)
+	}
+	return results, nil
+}
+
+func (o TestResultOutput) GetPrestoBucket(ctx context.Context, credentials evergreen.S3Credentials) (pail.Bucket, error) {
+	bucket, err := o.createPresto(ctx, credentials, o.BucketConfig.PrestoTestResultsPrefix, false)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating bucket")
+	}
+
+	return bucket, nil
+}
+
+// createPresto returns a Pail Bucket backed by PailType specifically for
+// buckets in our Presto ecosystem
+func (o TestResultOutput) createPresto(ctx context.Context, credentials evergreen.S3Credentials, prefix string, compress bool) (pail.Bucket, error) {
+	var b pail.Bucket
+	var stsConfig aws.Config
+	var err error
+
+	switch o.BucketConfig.Type {
+	case evergreen.BucketTypeS3:
+		stsConfig, err = config.LoadDefaultConfig(ctx,
+			config.WithRegion(evergreen.DefaultS3Region),
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "loading config")
+		}
+		stsConfig.Credentials = pail.CreateAWSStaticCredentials(credentials.Key, credentials.Secret, "")
+		stsClient := sts.NewFromConfig(stsConfig)
+		opts := pail.S3Options{
+			Name:        o.BucketConfig.PrestoBucket,
+			Prefix:      prefix,
+			Region:      evergreen.DefaultS3Region,
+			Permissions: pail.S3PermissionsPrivate,
+			Credentials: stscreds.NewAssumeRoleProvider(stsClient, o.BucketConfig.PrestoRoleARN),
+			MaxRetries:  utility.ToIntPtr(evergreen.DefaultS3MaxRetries),
+			Compress:    compress,
+			Verbose:     true,
```

Comment:
Do we want/care to have verbose logging enabled? I imagine it could add a good amount of noise to the Splunk logs.